lib/Doctrine/ORM/Mapping/AssociationOverrides.php

```diff
@@ -4,18 +4,42 @@
 
 namespace Doctrine\ORM\Mapping;
 
+use Attribute;
+
+use function is_array;
+
 /**
  * This annotation is used to override association mappings of relationship properties.
  *
  * @Annotation
+ * @NamedArgumentConstructor()
  * @Target("CLASS")
  */
+#[Attribute(Attribute::TARGET_CLASS)]
 final class AssociationOverrides implements Annotation
 {
     /**
      * Mapping overrides of relationship properties.
      *
      * @var array<\Doctrine\ORM\Mapping\AssociationOverride>
      */
-    public $value;
+    public $overrides = [];
+
+    /**
+     * @param array|AssociationOverride $overrides
```

Comment:
```suggestion
     * @param iterable<AssociationOverride>|AssociationOverride $overrides
```

Reply:
The code actually requires an array at the moment.

Reply:
Oh right, because of line 33, indeed!